src/benchq/resource_estimation/graph/graph_estimator.py

```diff
@@ -286,29 +289,35 @@ def estimate_resources_from_graph_data(
             algorithm_implementation.error_budget.transpilation_failure_tolerance
         )
 
+        widget_iterator = iter(self.widget_iterable_factory(self.hw_model))
+
         for this_transpilation_failure_tolerance in [
             transpilation_failure_tolerance * (0.1**i) for i in range(10)
         ]:
-            for widget in self.widget_iterator:
-                if "20-to-4" in widget.name:
-                    old_n_nodes = graph_data.n_nodes
-                    graph_data.n_nodes = ceil(graph_data.n_nodes / 4)
+            widget_found = False
+            for widget in widget_iterator:
+                tmp_graph_data = replace(
+                    graph_data,
+                    n_nodes=ceil(graph_data.n_nodes / widget.n_t_gates_produced),
+                )
```

Comment:
I love this! But can we change the name to `widget.n_t_gates_produced_per_cycle`?